packages/suite-storage/src/web/index.ts

```diff
@@ -7,33 +7,41 @@ import {
     IndexNames,
     IndexKey,
     StoreValue,
+    IDBPTransaction,
 } from 'idb';
 import { StorageUpdateMessage, StorageMessageEvent } from './types';
 
 // const VERSION = 1;
 
+export interface OnUpgradeProps<TDBStructure> {
+    db: IDBPDatabase<TDBStructure>;
+    oldVersion: number;
+    newVersion: number | null;
+    transaction: IDBPTransaction<TDBStructure, StoreNames<TDBStructure>[]>;
+}
+
 class CommonDB<TDBStructure> {
     private static instance: CommonDB<any>;
     dbName!: string;
     version!: number;
     db!: IDBPDatabase<TDBStructure> | null;
     broadcastChannel!: BroadcastChannel;
     onUpgrade!: (
-        db: IDBPDatabase<TDBStructure>,
-        oldVersion: number,
-        newVersion: number | null,
-        transaction: any
-    ) => void;
+        db: OnUpgradeProps<TDBStructure>['db'],
```

Comment:
wouldn't be cleaner to declare this function once?
Since you are using it here and then the same declaration in constructor?